src/common/types/objects/ledger_entries.ts

```diff
@@ -44,6 +45,72 @@ export interface CheckLedgerEntry {
   SourceTag: number
 }
 
+export interface DirectoryNodeLedgerEntry {
+  LedgerEntryType: 'DirectoryNode',
+  Flags: number,
+  RootIndex: number,
+  Indexes: string[],
+  IndexNext?: number,
+  IndexPrevious?: number
+}
+
+export interface OfferDirectoryNodeLedgerEntry
+       extends DirectoryNodeLedgerEntry {
+  Owner?: string,
+  TakerPaysCurrency?: string,
```

Comment:
I think all `TakerPays`/`TakerGets` fields are required in offer directory nodes.
Same for `Owner` in owner directories.